### PR TITLE
Clean up `suspend(while:)`

### PR DESCRIPTION
**Pull request by stephencelis**
Realizing we can just lean on the new `Publisher.values` bridge to simplify things.

### CODE REVIEW

**Review by mbrandonw — APPROVED** [review]
lol wow. much better

File: Sources/ComposableArchitecture/Beta/Concurrency.swift [hunk 1]
```diff
@@ -218,35 +218,10 @@ import SwiftUI
...
-  private class Box<Value> {
-    var wrappedValue: Value
-
-    init(wrappedValue: Value) {
-      self.wrappedValue = wrappedValue
+      _ = await self.publisher
+        .values
+        .filter { !predicate($0) }
+        .first(where: { _ in true })
     }
```
Comment (iampatbrown) [inline Sources/ComposableArchitecture/Beta/Concurrency.swift:225]: Nice thinking! Could also skip the filter if you wanted.
```swift
 _ = await self.publisher
        .values
        .first(where: { !predicate($0) })
```

Comment (stephencelis) [reply Sources/ComposableArchitecture/Beta/Concurrency.swift:225]: Ha good call! Wanna do a quick PR?

Comment (iampatbrown) [reply Sources/ComposableArchitecture/Beta/Concurrency.swift:225]: No worries #730

**Pull request merged**